crates/task-impls/src/quorum_vote/handlers.rs

```diff
@@ -175,26 +173,23 @@ pub(crate) async fn handle_quorum_proposal_validated<
                 .block_number();
 
             // Skip if this is not the expected block.
-            if task_state.epoch_height != 0
-                && (decided_block_number + 3) % task_state.epoch_height == 0
-            {
+            if task_state.epoch_height != 0 && decided_block_number % task_state.epoch_height == 0 {
```

Comment:
This part is inside the `if let Some(decided_view_number) = new_decided_view_number` clause, but we actually want to start the computation no matter if we have decided on a new view, so it should be moved to the beginning of this function.

Reply:
Besides, we probably don't have to rely on `decided_block_number % task_state.epoch_height == 0`, since it's possible that we somehow miss the first block. Instead, we can check if `task` has an in-progress computation task for the next epoch, and if not, we start the computation.